### PR TITLE
fix kubeadm ci path logic

### PR DESCRIPTION
**Pull request by BenTheElder**
- drop dead 1.6 support
- correct some paths in kubetest and kubernetes_e2e.py related to kubernetes anywhere / kubeadm

Note CI bazel builds are now at EG:
```
$ gsutil cat gs://kubernetes-release-dev/ci/latest-bazel.txt
v1.11.0-alpha.0.1031+d37460147ec956-bazel
```
Fixing presubmits will come separately. We'll also need to bump kubetest and we should reconfigure all of the jobs to use the `latest-bazel(-release1.x)?` tags as well as no `run_after_success`.

### CODE REVIEW

**Comment by BenTheElder** [comment]
looks like I need to update some tests

File: scenarios/kubernetes_e2e.py [hunk 1]
```diff
@@ -136,14 +136,8 @@ def kubeadm_version(mode, shared_build_gcs_path):
...
-        # Work-around for release-1.6 jobs, which still upload debs to an older
-        # location (without os/arch prefixes).
-        # TODO(pipejakob): remove this when we no longer support 1.6.x.
-        if version.startswith('v1.6.'):
-            return 'gs://kubernetes-release-dev/bazel/%s/build/debs/' % version
-
-        # The path given here should match jobs/ci-kubernetes-bazel-build.sh
-        return 'gs://kubernetes-release-dev/bazel/%s/bin/linux/amd64/' % version
+        # The path given here should match ci-kubernetes-bazel-build
+        return 'gs://kubernetes-release-dev/%s-bazel/bin/linux/amd64/' % version
```
Comment (jessicaochen) [inline scenarios/kubernetes_e2e.py]: Are you missing a ci in this path?

Comment (BenTheElder) [reply scenarios/kubernetes_e2e.py]: yes

Comment (BenTheElder) [reply scenarios/kubernetes_e2e.py]: I have this fixed locally, still looking at all the test failures

Comment (BenTheElder) [reply scenarios/kubernetes_e2e.py]: done

**Comment by BenTheElder** [comment]
PTAL, corrected some more paths in the tests especially, dropped 1.6 related tests (we don't support kubernetes 1.6).

**Comment by krzyzacy** [comment]
looks fine to me, will let @jessicaochen stamp

**Comment by BenTheElder** [comment]
we'll need to bump kubetest
/hold

**Comment by BenTheElder** [comment]
creating the bump PR

**Comment by jessicaochen** [comment]
Is is possible to also kick the build for bazel pre 1.8? The latest version for those are still not defined.

gsutil ls gs://kubernetes-release-dev/ci/latest-bazel*

gs://kubernetes-release-dev/ci/latest-bazel-1.10.txt
gs://kubernetes-release-dev/ci/latest-bazel-1.9.txt
gs://kubernetes-release-dev/ci/latest-bazel.txt

**Comment by BenTheElder** [comment]
I'm migrating these to periodics as well in the config fix, which should
solve that.

**Comment by BenTheElder** [comment]
FWIW i'm also unhappy that there are so many assumed paths in here tied to how we upload bazel, but I think getting things onto a build compatible with our other CI builds upload format is important and this should let us start improving the scheduling etc. and avoid downloading the wrong versions.

**Pull request merged**